### PR TITLE
enrolment list api changes to support the resume progress

### PR DESCRIPTION
**Pull request by shankaragoudab**

### CODE REVIEW

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/JsonKey.java [hunk 1]
```diff
@@ -1136,6 +1136,7 @@ public final class JsonKey {
   public static final String USER_KARMA_TOTAL_POINTS = "total_points";
   public static final String ADD_INFO = "addinfo";
   public static final String USERID_LIST="userIdList";
+  public static final String LRC_PROGRESS_DETAILS = "lrc_progressdetails";
```
Comment (karthik-tarento) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/JsonKey.java]: use the same in cassandratablecolumn.properties... with camel case string

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala [hunk 2]
```diff
@@ -398,6 +399,13 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
             val progress: Int = enrolment.getOrDefault("progress", 0.asInstanceOf[AnyRef]).asInstanceOf[Int]
             enrolment.put("status", getCompletionStatus(progress, leafNodesCount).asInstanceOf[AnyRef])
             enrolment.put("completionPercentage", getCompletionPerc(progress, leafNodesCount).asInstanceOf[AnyRef])
+            val lrc_progress: String = enrolment.get(JsonKey.LRC_PROGRESS_DETAILS).asInstanceOf[String]
```
Comment (karthik-tarento) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: use option and assign empty string if this is null

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala [hunk 3]
```diff
@@ -398,6 +399,13 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
...
+            var mappedValue: util.Map[String, AnyRef] = new util.HashMap[String, AnyRef]()
+            if (lrc_progress != null) {
+                val objectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
+                mappedValue = objectMapper.readValue(lrc_progress, classOf[util.Map[String, AnyRef]])
+            }
```
Comment (karthik-tarento) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: use try and get or else -- which assigns empty map...

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala [hunk 4]
```diff
@@ -159,7 +159,8 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
         logger.info(request.getRequestContext,"CourseEnrolmentActor :: list :: UserId = " + userId)
         try{
             val response = if (isCacheEnabled && request.getContext.get("cache").asInstanceOf[Boolean])
-                getCachedEnrolmentList(userId, () => getEnrolmentList(request, userId, courseIdList)) else getEnrolmentList(request, userId, courseIdList)
+                getCachedEnrolmentList(userId, () => getEnrolmentList(request, userId, courseIdList))
+            else getEnrolmentList(request, userId, courseIdList)
```
Comment (karthik-tarento) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: remove this formating

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala (hunk 2 above)
Comment (shankaragoudab) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: updated

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala (hunk 3 above)
Comment (shankaragoudab) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: updated

**Pull request merged**